### PR TITLE
avoid mutex locking around blocking calls in kubelet-executor

### PR DESCRIPTION
**Pull request by jdef**
xref https://github.com/mesosphere/kubernetes-mesos-private/issues/63

### CODE REVIEW

**Comment by jdef** [comment]
@sttts @s-urbaniak PTAL

**Comment by s-urbaniak** [comment]
lgtm

**Comment by jdef** [comment]
thanks. building an image for testing now. not expecting problems..

File: contrib/mesos/pkg/executor/service/service.go [hunk 1]
```diff
@@ -45,8 +44,8 @@ type KubeletExecutorServer struct {
 	SuicideTimeout    time.Duration
 	LaunchGracePeriod time.Duration
 
-	kletLock sync.Mutex // TODO(sttts): remove necessity to access the kubelet from the executor
-	klet     *kubelet.Kubelet
+	klet      *kubelet.Kubelet // once set, immutable
```
Comment (sttts) [inline contrib/mesos/pkg/executor/service/service.go]: please add the TODO again. This is still valid. All this synchronization should not be necessary if we had decoupled both components.

**Comment by sttts** [comment]
When my comment is addressed, lgtm.

**Comment by jdef** [comment]
thanks for the fast reviews guys. i added back the comment, squashed and re-pushed

**Comment by s-urbaniak** [comment]
:+1:

**Pull request merged**